api_app/helpers.py

```diff
@@ -64,3 +65,21 @@ def get_binary(job_id, job_object=None):
 def generate_sha256(job_id):
     binary = get_binary(job_id)
     return hashlib.sha256(binary).hexdigest()
+
+
+def get_hash_type(hash_value):
+    """
+    Returns hash type
+    Supports md5, sha1, sha256 and sha512
+    """
+    RE_HASH_MAP = {
+        "md5": re.compile(r"^[a-f\d]{32}$", re.IGNORECASE | re.ASCII),
+        "sha1": re.compile(r"^[a-f\d]{40}$", re.IGNORECASE | re.ASCII),
+        "sha256": re.compile(r"^[a-f\d]{64}$", re.IGNORECASE | re.ASCII),
+        "sha512": re.compile(r"^[a-f\d]{128}$", re.IGNORECASE | re.ASCII),
+    }
+
+    for hash_type, re_hash in RE_HASH_MAP.items():
+        if re.match(re_hash, hash_value):
+            return hash_type
+    return None  # no matches
```

Comment:
In this function, having 2 different `return` code is not a particular problem because the function is short and we all get what is happening.

Anyway, it would be cleaner to just use `return` only a single time in the code. In this way it is easier to read by other programmers because there is no hidden behavior that could led to return the function before expected.

So, in this case, you can just use a variable, change it when you find a match and then return that variable only a single time at the end of the code.